sensor/kubernetes/crs/crs.go

```diff
@@ -0,0 +1,341 @@
+package crs
+
+import (
+	"context"
+	"crypto/tls"
+	"fmt"
+	"os"
+	"path/filepath"
+	"time"
+
+	metautils "github.com/grpc-ecosystem/go-grpc-middleware/v2/metadata"
+	"github.com/pkg/errors"
+	"github.com/stackrox/rox/generated/internalapi/central"
+	"github.com/stackrox/rox/generated/storage"
+	"github.com/stackrox/rox/pkg/booleanpolicy/policyversion"
+	"github.com/stackrox/rox/pkg/centralsensor"
+	"github.com/stackrox/rox/pkg/clientconn"
+	"github.com/stackrox/rox/pkg/clusterid"
+	"github.com/stackrox/rox/pkg/crs"
+	"github.com/stackrox/rox/pkg/env"
+	grpcUtil "github.com/stackrox/rox/pkg/grpc/util"
+	"github.com/stackrox/rox/pkg/k8sutil"
+	"github.com/stackrox/rox/pkg/logging"
+	"github.com/stackrox/rox/pkg/mtls"
+	"github.com/stackrox/rox/pkg/pods"
+	protoconv "github.com/stackrox/rox/pkg/protoconv/certs"
+	"github.com/stackrox/rox/pkg/protoutils"
+	"github.com/stackrox/rox/pkg/version"
+	"github.com/stackrox/rox/sensor/common/centralclient"
+	"github.com/stackrox/rox/sensor/common/sensor/helmconfig"
+	"github.com/stackrox/rox/sensor/kubernetes/certrefresh"
+	"github.com/stackrox/rox/sensor/kubernetes/certrefresh/securedcluster"
+	"github.com/stackrox/rox/sensor/kubernetes/sensor"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/encoding/gzip"
+	"google.golang.org/grpc/metadata"
+	"google.golang.org/grpc/status"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/kubernetes"
+)
+
+const crsTempDirEnvVarName = "ROX_CRS_TMP_DIR"
+
+var (
+	log                                  = logging.LoggerForModule()
+	fetchSensorDeploymentOwnerRefBackoff = wait.Backoff{
+		Duration: 10 * time.Millisecond,
+		Factor:   3,
+		Jitter:   0.1,
+		Steps:    10,
+		Cap:      6 * time.Minute,
+	}
+)
+
+// EnsureClusterRegistered initiates the CRS based cluster registration flow in case a
+// CRS is found instead of regular service certificate.
+func EnsureClusterRegistered() error {
+	log.Infof("Ensuring Secured Cluster is registered.")
+	clientconn.SetUserAgent(fmt.Sprintf("%s CSR", clientconn.Sensor))
+
+	// Check if we service certificates are missing.
+	_, err := mtls.LeafCertificateFromFile()
```

Comment:
@vladbologa, since CRS-retrieved certs will be persisted in the form of K8s secrets named `tls-*`, we should be able to make the logic depend on the existence of e.g. a K8s secret named `tls-sensor`, right? If that secret exists, we can assume that CRS-based registration has already happened and we could return early (successfully) from `EnsureClusterRegistered`?

Reply:
See commit f92668a73209a784597cf4063571853e311e8f33.